### ISSUE TITLE
Sequential.summary(): keras-style structural summary with shape inference

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/xrtoolz#89]
## Motivation

PR #88 ships a tree-style `Sequential.describe()`:

```
Sequential (3 ops)
├── ValidateCoords()
├── SubsetBBox(lon_bnds=[-125, -65], lat_bnds=[25, 50], lon='lon', lat='lat')
└── ResampleTime(freq='1D', method='mean', time='time')
```

That's the maximum we can do from `get_config()` alone. To match `keras.Model.summary()` properly we need **structural** info — input/output shape per step. That requires a shape-inference protocol on every `Operator`.

## Proposal

Add `Operator.compute_output_signature(input_signature) -> output_signature` (analogous to keras's `compute_output_spec`). The signature is a thin descriptor — `dict[dim_name, size | None]` plus dtype — that propagates through a `Sequential` / `Graph` without executing the real data path.

Then add `Sequential.summary(input_signature)` rendering an input/output shape per step, à la keras.

## Scope

In-scope:
- A `Signature` dataclass.
- `Operator.compute_output_signature` default that returns input unchanged.
- Override on every shipped Operator that changes shape (subset / resample / regrid / metric / select / rename).
- `Sequential.summary(signature)` and `Graph.summary({name: signature})`.
- Tests covering each override + a Sequential roll-up test.

Out-of-scope:
- Auto-inferring signatures from a real Dataset (sugar later: `pipeline.summary_from(ds)`).
- Symbolic dim arithmetic — only render strings, not algebra.

## Tradeoffs

- Touches every Operator subclass — that's why this is its own issue, not a quick PR.
- Without it, `describe()` will always be limited to what's in `get_config()`.
- With it, `xr_toolz` gets a debugging surface much closer to keras and can also catch shape mismatches at compile time inside `Graph`.

## Related

- PR #88 — tree-style `describe()` (visible-quick-win precursor).
